src/components/WorkflowEngine/poller.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * Calls a function passed with a specified delay, allows to cancel using the
+ * return function.
+ * @param fn Function that is invoked after each delay.
+ */
+export const poller: PollerFunc = (fn) => {
+  let timeout: ReturnType<typeof setTimeout> | null = null
+  let isCancelled = false
+
+  const poll: PollFunc = (delay) => {
+    if (timeout) clearTimeout(timeout)
+
+    timeout = setTimeout(() => {
+      if (isCancelled) return
+      fn(poll)
+    }, delay)
+  }
+
+  const cancel: CancelFunc = () => {
+    if (isCancelled) return
+
+    if (timeout) clearTimeout(timeout)
+
+    timeout = null
+    isCancelled = true
+  }
+
+  fn(poll)
+
+  return cancel
```

Comment:
Cancel is not used at all beside the tests. We might want to get ease the function here.

Reply:
we are going to have a retry functionality in Q2. It is future planing ...